src/coreclr/src/tools/crossgen2/crossgen2/Program.cs

```diff
@@ -242,6 +244,18 @@ private int Run()
                         try
                         {
                             EcmaModule module = typeSystemContext.GetModuleFromPath(referenceFile);
+                            if (versionBubbleModulesHash.Contains(module))
+                            {
+                                // Ignore reference assemblies that have also been passed as inputs
+                                continue;
+                            }
+                            if (_commandLineOptions.Composite)
+                            {
+                                // In composite mode, add reference assemblies as inputs but don't root all their methods
+                                // This will need more granularity (probably two types of references) when implementing
+                                // shared framework / multi-layer composite R2R files.
+                                inputModules.Add(module);
```

Comment:
It seems very strange that we'd want to include in the input set any modules that were only specified as references. I don't see why you are doing this. Instead I would expect to see the logic that changed what were the reference vs input modules to be placed into SuperILC

Reply:
As I described elsewhere, this was more like a first shot at something like a potential rooting policy. Rooting the entire framework is simply impractical for testing as the build takes ages and the output file is huge; the limited rooting makes the build much easier to manage, at least for the purpose of local testing. If you think there's no value in having this knob, I can easily keep this as local instrumentation. In general I'm however inclined to think that a "second type of input file" meaning component assembly that is included in the file but not automatically rooted would be generally helpful. If we agree on implementing this option, I can easily modify SuperIlc to use it and drop this temporary hack. After all, I'm starting to get consistent R2RDump info for composite images so I believe I'm ready to dive into runtime implementation and after that the next step should be layered composite images I guess.